flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/DispatcherResourceCleanerFactory.java

```diff
@@ -101,13 +101,13 @@ public ResourceCleaner createLocalResourceCleaner(
     @Override
     public ResourceCleaner createGlobalResourceCleaner(
             ComponentMainThreadExecutor mainThreadExecutor) {
-
         return DefaultResourceCleaner.forGloballyCleanableResources(
                         mainThreadExecutor, cleanupExecutor, retryStrategy)
-                .withPrioritizedCleanup(jobManagerRunnerRegistry)
+                .withPrioritizedCleanup(ofLocalResource(jobManagerRunnerRegistry))
                 .withRegularCleanup(jobGraphWriter)
```

Comment:
I don't understand why we need to replicate things here.

Shouldn't every local resources always be cleaned up on a global cleanup? Why can we not, in the case where currently we'd use the global cleaner, first call the local cleaner and then run the global one?

Is there any use-case where a local resource should _not_ be cleaned up on a global terminal state?
Is there any use-case where a local resource should be cleaned _after_ global resource?